src/Symfony/Component/Dotenv/Dotenv.php

```diff
@@ -50,6 +51,33 @@ public function load(string $path, string ...$paths): void
         $this->doLoad(false, $path, $paths);
     }
 
+    /**
+     * Loads one or several .env and the corresponding env.local, env.$env and env.$env.local files if they exist.
+     *
+     * .env.local is always ignored in test env because tests should produce the same results for everyone.
+     *
+     * @param string    $path  A file to load
+     * @param ...string $paths A list of additional files to load
+     *
+     * @throws FormatException when a file has a syntax error
+     * @throws PathException   when a file does not exist or is not readable
+     *
+     * @see https://github.com/bkeepers/dotenv#what-other-env-files-can-i-use
+     */
+    public function loadForEnv(string $env, string $path, string ...$paths): void
+    {
+        array_unshift($paths, $path);
+        foreach ($paths as $p) {
+            $pathList = array("$p.$env");
+            if ('test' !== $env) {
```

Comment:
This hardcode is a convention that must be documented and memorized. Could it be an argument `array $exclude = array('test')` for example? Configuration over convention is always better and helps to understand what will happen inside and be flexible for edge cases.

Reply:
See https://github.com/symfony/symfony/pull/28533#discussion_r219424365
All existing tools already follow this convention, I don't want to add something Symfony-specific here.

Reply:
@dunglas What if my testing environment is not called `test`?

Reply:
Then you’re on your own. The tool will not prevent you from shooting yourself in the foot.
It’s why using standards and widespread conventions is interesting.

If you really want to make this configurable, I can add a constructor parameter, but I would prefer not doing that.